docs/source/developer_guide/release.rst

```diff
@@ -3,32 +3,106 @@
-you can follow. Simply run:
+
+The release process is as follows, on high-level:
+
+1. ensure deprecation actions are carried out.
+  Deprecation actions for a version should be marked by "version number" annotated comments in the code.
+  E.g., for the release 0.12.0, search for the string 0.12.0 in the code and carry out described deprecation actions.
+  Collect list of deprecation actions in an issue, as they will have to go in the release notes.
+  Deprecation actions should be merged only by release managers.
+
+2. create a "release" pull request (ideally from a branch following the naming pattern ``release/v0.x.y``). This should make changes to the version numbers and have complete release notes.
+  See below for version numbers and release notes.
+
+3. The PR and release notes should optimally be reviewed by the other core developers, then merged once tests pass.
+
+4. create a GitHub draft release with a new tag following the syntax v[MAJOR].[MINOR].[PATCH],
+   e.g., the string ``v0.12.0`` for version 0.12.0.
+   The GitHub release notes should contain only "new contributors" and "all contributors" section,
+   and otherwise link to the release notes in the changelog,
+   following the pattern of current GitHub release notes.
+
+5. publish the GitHub draft release. Creation of the new tag will trigger the release CI/CD.
+
+6. wait for the release CI/CD to finish. If tests fail due to sporadic unrelated failure, restart.
+  If tests fail genuinely, something went wrong in the above steps, investigate, fix, and repeat.
+  Common possibilities are core devs not respecting the feature freeze period,
+  new releases of dependencies that happen in the ca one day period between release PR and release.
+
+7. once release CI/CD has passed, check ``sktime`` version on ``pypi``, this should be the new version.
+  It should be checked that all wheels have been uploaded, `here <https://pypi.org/simple/sktime/>`__.
+  A validatory install of ``sktime`` in a new ``python`` environment should be carried out (one arbitrary version/OS),
+  according to the install instructions.
+  If the install does not succeed or wheels have not been uploaded, urgent action to diagnose and remedy must be taken.
+  All core developers should be urgently informed of such a situation through mail-all in the core developer channel on slack.
+  In the most common case, the install instructions need to be updated.
+
+8. If the release on ``pypi`` has succeeded, there should be an automated release PR created  against the sktime conda-forge repo: https://github.com/conda-forge/sktime-feedstock. This PR need to be reviewed and in dependencies should be checked against any changes in the main sktime repo. In case the dependencies (or python version support) have changes the ``meta.yml`` file in the conda recipe need to updated to reflect those changes. Once the PR is merge tit will trigger a release of the conda package.  
```

Comment:
couple strange sentences in your addition, @lmmentel - can you check and fix, please?